### DISCUSSION TITLE
Why does the server send a ping packet from version 3?

### DISCUSSION THREAD

**Discussion by evgencode** [socketio/socket.io#4094] [Q&A] [answered]
Why does the server send a ping packet from version 3?
In the second version, the ping packet was sent by the client.
I think it would be more correct to send a ping from the client

**Comment by darrachequesne** [comment, marked as answer]
Please check the reasoning here: https://socket.io/blog/engine-io-4-release/#heartbeat-mechanism-reversal

> After analysis, it seems to be caused by delayed timers on the client-side. Those timers are used in the ping-pong mechanism which helps to ensure the connection between the server and the client is still healthy. A delay on the client-side meant the client sent the ping packet too late, and the server considered that the connection was closed.
>
> That's why the ping packets will now be sent by the server, and the client will respond with a pong packet.
